flow-nodes/api-builder-plugin-format-date/src/flow-nodes.yml

```diff
@@ -5,32 +5,35 @@ flow-nodes:
     description: Example flow-node that formats Date. Uses the popular Moment.js library.
     category: general
     methods:
-      formatDate:
+      formatDate: # This same key is used in `actions.js`
         name: Format Date
         description: Formats a date. See Moment.js format() method.
         parameters:
           date:
+            name: Date
             description: The date to be formatted.
             required: true
             initialType: string
             schema:
               type: string
           format:
+            name: Format
             description: The desired format. Supports all Moment.js supported formats.
             required: true
             initialType: string
             schema:
               type: string
-        outputs:
-          # success output(s) should be first
-          next:
-            name: Next
-            context: $.date
-            schema:
-              type: string
-          # error output(s) should be last
-          error:
-            name: Error
-            context: $.error
-            schema:
-              type: string
+        # "Next" (next) output
+        returns:
+          name: Next
+          description: The operation was successful.
+          context: $.date
+          schema:
+            type: string
+        # "Error" (error) output
+        throws:
+          name: Error
+          description: An unexpected error was encountered.
```

Comment:
There's a lot of cases when you'd get the error output and they weren't unexpected. Similar to returns, you should give a description relevant to each flow-node and method (same for all plugins)

Reply:
@alasdairhurst, I'm not sure I agree here.  It's unexpected for good reason - we can't predict what the errors are going to be.  There's one error output for all errors.  If you think you can provide a better description, please suggest one.

Reply:
We know for sure that the output is going to be triggered for parameter validation issues in this plugin, for example. Because we throw errors there.
Another example, the javascript one, triggers the error output when the code that is provided throws or rejects.
Another example could be network, auth or service connectivity issues with the redis plugin. 

Known errors can be listed here depending on how limited the cases are, and how common you will see errors. (if there's 2 known cases, list them, if there's 50 then obviously don't list them all, list the common ones and leave the rest up to the readme or something :P)

In this case for this plugin, you should only ever see errors in the cases that we handle (unless there was some sort of system issue) so we can say something along the lines of `Invalid parameters were provided, or an unexpected error occurred`.

In a case that could have a lot of errors just mention the more common ones and stick `or an unexpected error occurred` on the end imo. Gives the plugins something tailored to them. Otherwise, if we just want the same thing for error and have a good reason for it, we may as well just hardcode it in the SDK instead of making everyone define the string...

Reply:
It's an impossible and somewhat fruitless / pointless task to try to identify errors that could literally be caused by anything, including permissions, disk space, bugs, etc.  It is documentation that would be grossly inaccurate and insufficient in detail (much less so than the content of the _actual_ errors).  So, no, I don't think an _example_ flow-node really needs this kind of attention to detail.  I believe "An unexpected error was encountered." is suitably covers all eventualities.

Reply:
It's not just the examples that are setting this description though, it's most of our flow-nodes that have been updated if i remember correctly